docs/how-tos/rust_service.md

```diff
@@ -0,0 +1,334 @@
+- `axum` as the HTTP server
+- `async-graphql` as graphql server library 
+- `async-graphql-axum` for GraphQL request handling
+- `tokio` as runtime for writing asynchronous rust program
+
+```toml
+[dependencies]
+async-graphql = "7.0.6"
+async-graphql-axum = "7.0.6"
+axum = "0.7.5"
+tokio = { version = "1.38.0", features = ["macros", "rt-multi-thread"] }
+```
+
+## Serving GraphQL Requests
+
+To handle GraphQL requests, we start by creating a server with a placeholder GraphQL schema. We use the `async-graphql-axum` crate, which provides integration between `async-graphql` and `axum`, to create a `graphql_handler`.
+
+!!! example "GraphQL Request Handler"
+
+    ```rust
+    use async_graphql::{EmptyMutation, EmptySubscription, Object, Schema};
+    use async_graphql_axum::{GraphQLRequest, GraphQLResponse};
+    use axum::{
+        extract::Extension,
+        routing::post,
+        Router,
+    };
+
+    struct Query;
+
+    #[tokio::main]
+    async fn main() {
+        let schema = Schema::build(Query, EmptyMutation, EmptySubscription).finish();
+
+        let app = Router::new()
+            .route("/graphql", post(graphql_handler))
+            .layer(Extension(schema));
+
+        let listener = tokio::net::TcpListener::bind("127.0.0.1:8000").await.unwrap();
+        axum::serve(listener, app).await.unwrap();
+    }
+
+    async fn graphql_handler(
+        schema: Extension<Schema<Query, EmptyMutation, EmptySubscription>>,
+        req: GraphQLRequest,
+    ) -> GraphQLResponse {
+        schema.execute(req.into_inner()).await.into()
+    }
+    ```
+
+!!! info
+
+    `GraphQLRequest` handles incoming GraphQL requests. It wraps the raw HTTP request
+    data and converts it into a form that can be processed by async-graphql.
+
+    `GraphQLResponse` wraps the results of executing a GraphQL query. It ensures
+    that the response is correctly formatted as a valid HTTP response that Axum can 
+    send back to the client.
+
+## Defining Objects
+
+!!! info
+
+    Refer to [Async-graphql Book SimpleObject](https://async-graphql.github.io/async-graphql/en/define_simple_object.html)
+    for a full explanation of how to implement various objects and resolvers.
+
+An `Object` can be defined by using the  `Object` macro from `async-graphql` crate. A GraphQL object must have
+a resolver defined for each field in its `impl`.
+
+We can use the `SimpleObject` macro from `async-graphql` to map all the fields of a struct to a GraphQL object.
+
+```rust
+use async_graphql::SimpleObject;
+#[derive(SimpleObject)]
+struct Person {
+    id: u32, 
+    first_name: String, 
+    last_name: String, 
+    preferred_name: Optional<String>,
+}
+```
+We can write a resolver for the `Person`. A resolver function resolvers the values for all the fields a GraphQL object.
+
+!!! example "Defind Object and a Resolver"
+
+    ```rust
+    use async_graphql::Object;
+    struct Query;
+
+    #[Object]
+    impl Query {
+        async fn get_person(&self) -> Person {
```

Comment:
The prefix `get_` isn't used in GraphQL APIs as they're already known to be Queries

```suggestion
        async fn person(&self) -> Person {
```